platform/MCU/shared/STM32/src/hw_ticks.c

```diff
@@ -66,7 +66,9 @@ uint64_t GetSystem1MsTick64()
     const int is = __get_PRIMASK();
     __disable_irq();
 
-    millis = system_millis + (DWT->CYCCNT - system_millis_clock) / SYSTEM_US_TICKS / 1000;
+    system_tick_t elapsedTicks = (DWT->CYCCNT >= system_millis_clock) ? (DWT->CYCCNT - system_millis_clock) : (sizeof(uint32_t) - system_millis_clock + DWT->CYCCNT);
```

Comment:
This is not going to work as intended. CYCCNT is a 32 bit counter that ticks at 120MHz and thus wraps every ~30 seconds. The check for `DWT->CYCNT >= system_millis_clock` doesn't really tell us anything.

Resetting `system_millis_clock` to current `DWT->CYCCNT` on wakeup should be sufficient.